docs/sequence.rst

```diff
@@ -0,0 +1,672 @@
+Sequences
+=========
+
+.. highlight:: haskell
+
+Sequences allow you to store a finite number of sequential elements, providing
+fast access to both ends of the sequence as well as efficient concatenation. The
+``containers`` package provides the `Data.Sequence
+<https://hackage.haskell.org/package/containers/docs/Data-Sequence.html>`_ module
+which defines the ``Seq`` data type.
+
+
+Short Example
+-------------
+
+The following GHCi session shows some of the basic sequence funcitonality.
+
+.. code-block:: haskell
+
+    -- Import the Seq type and operators for combining sequences unqualified.
+    -- Import the rest of the Sequence module qualified.
+    import Data.Sequence (Seq(..), (<|), (|>), (><))
+    import qualified Data.Sequence as Seq
+
+    let nums = Seq.fromList [1, 2, 3]
+
+
+    -- Put numbers on the front and back.
+    0 <| nums
+    > fromList [0,1,2,3]
+
+    nums |> 4
+    > fromList [1,2,3,4]
+
+
+    -- Reverse a sequence
+    Seq.reverse (Seq.fromList [0, 1, 2])
+    > fromList [2,1,0]
+
+
+    -- Put two sequences together.
+    (Seq.fromList [-2, -1]) >< nums
+    > fromList [-2,-1,0,1,2]
+
+
+    -- Check if a sequence is empty and check the length.
+    Seq.null nums
+    > False
+
+    Seq.length nums
+    > 3
+
+
+    -- Lookup an element at a certain index (since version 0.4.8).
+    Seq.lookup 2 nums
+    > Just 3
+
+    -- Or the unsafe version, you MUST check length beforehand.
+    Seq.index 2 nums
+    > 3
+
+
+    -- Map a function over a sequence (can use fmap or the infix function <$>).
+    fmap show nums
+    > fromList ["0","1","2"]
+
+    show <$> nums
+    > fromList ["0","1","2"]
+
+
+    -- Fold a sequence into a summary value.
+    foldr (+) 0 (Seq.fromList [0, 1, 2])
+    > 3
+
+.. TIP:: You can use the `OverloadedLists
+	 <https://ghc.haskell.org/trac/ghc/wiki/OverloadedLists>`_ extension so
```

Comment:
Why are you linking to the GHC developer's wiki rather than the documentation?

Reply:
The real documentation is here: https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#overloaded-lists

Reply:
Probably because its neither on the first or second page when googling for "overloadedlists haskell". I didn't even know that page existed :worried:.

*/me adds this to the list of examples of "impossible to find documentation".*